galleon-pack/src/main/resources/packages/docs.examples/content/docs/examples/enable-microprofile.cli

```diff
@@ -0,0 +1,59 @@
+# This CLI script allows to enable Microprofile for the standalone configurations.
+# By default, standalone.xml is updated.
+# Run it from JBOSS_HOME as:
+# bin/jboss-cli.sh --file=docs/examples/enable-microprofile.cli [-Dconfig=<standalone-full.xml|standalone-ha.xml|standalone-full-ha.xml>]
+
+embed-server --server-config=${config:standalone.xml}
+
+echo INFO: Updating configuration to use elytron
+
+/subsystem=undertow/application-security-domain=other:add(security-domain=ApplicationDomain)
+/subsystem=ejb3/application-security-domain=other:add(security-domain=ApplicationDomain)
+/subsystem=batch-jberet:write-attribute(name=security-domain, value=ApplicationDomain)
+
+# This applies to full configurations only.
+if (outcome == success) of /subsystem=messaging-activemq/server=default:read-resource
+  /subsystem=messaging-activemq/server=default:undefine-attribute(name=security-domain)
+  /subsystem=messaging-activemq/server=default:write-attribute(name=elytron-domain, value=ApplicationDomain)
+end-if
+
+/subsystem=remoting/http-connector=http-remoting-connector:write-attribute(name=sasl-authentication-factory, value=application-sasl-authentication)
+/subsystem=remoting/http-connector=http-remoting-connector:undefine-attribute(name=security-realm)
+
+/core-service=management/access=identity:add(security-domain=ManagementDomain)
+/core-service=management/management-interface=http-interface:write-attribute(name=http-upgrade,value={enabled=true, sasl-authentication-factory=management-sasl-authentication})
+/core-service=management/management-interface=http-interface:write-attribute(name=http-authentication-factory,value=management-http-authentication)
+/core-service=management/management-interface=http-interface:undefine-attribute(name=security-realm)
+
```

Comment:
What about slimming the ApplicationRealm so it is just about default SSL, as Darran suggested in the analysis.

Reply:
@OndrejKotek , the http-invoker references the ApplicationRealm, to be able to remove the authentication part we would need to introduce an http-authentication factory as defined in the analysis: https://github.com/wildfly/wildfly-proposals/pull/294/files#diff-f4f1c6de1cff1faeef00be5fbe9ef028R82